internal/repo/client/clientset.go

```diff
@@ -22,24 +23,40 @@ type defaultClientsetClient struct {
 
 type defaultClientset struct {
 	client            defaultClientsetClient
-	clients           map[string]RepoClient
+	clients           map[string]repoClientWithState
+	clientAge         map[string]time.Time
 	repoWithNameMutex sync.Mutex
 	repoMutex         sync.Mutex
 	maxCacheAge       time.Duration
 }
 
 var _ RepoClientset = &defaultClientset{}
 
+type repoClientWithState struct {
+	client              RepoClient
+	lastCheckedRepoSpec time.Time
+	repo                v1alpha1.PackageRepository
+}
+
+func (s *repoClientWithState) lastCheckedRepoSpecAgo(maxAge time.Duration) bool {
+	return s.lastCheckedRepoSpec.Add(maxAge).After(time.Now())
+}
+
+func (s *repoClientWithState) checkRepoSpec(repo v1alpha1.PackageRepository) bool {
+	return reflect.DeepEqual(s.repo.Spec, repo.Spec)
+}
+
 func NewClientset(pkgClient adapter.PackageClientAdapter, k8sClient adapter.KubernetesClientAdapter) RepoClientset {
-	return NewClientsetWithMaxCacheAge(pkgClient, k8sClient, 5*time.Minute)
+	return NewClientsetWithMaxCacheAge(pkgClient, k8sClient, 30*time.Second)
```

Comment:
IMHO 30 seconds is quite short. I know 5 minutes on the other hand is quite long. I think we should do #756.

Reply:
Ah I missed that this value is also used for invalidating the cache for the repo client (not just the repo info). I think it would make sense to allow specifying different values for both. I think 30s is reasonable for the repo info but definitely on the shorter side for the repo client

Reply:
As for #756, I think it should be tackled in a separate PR as it seems kind of unrelated

Reply:
updated, let me know what you think

Reply:
> As for https://github.com/glasskube/glasskube/issues/756, I think it should be tackled in a separate PR as it seems kind of unrelated

yes, that's what i meant